lib/Table.js

```diff
@@ -515,56 +544,71 @@ const buildTableReq = Table.prototype.buildTableReq = function buildTableReq(nam
 };
 
 Table.prototype.create = function (next) {
-  let ddb = this.base.ddb();
-  let schema = this.schema;
-  let createTableReq = this.buildTableReq(this.name, schema);
 
-  debug('ddb.createTable request:', createTableReq);
+  if(next){
+    this.create()
+      .then(result => {next(undefined,result)}, err => {next(err)})
+    return
+  }
 
-  const deferred = Q.defer();
+  return new Promise((resolve, reject) => {
+    let ddb = this.base.ddb();
+    let schema = this.schema;
+    let createTableReq = this.buildTableReq(this.name, schema);
 
-  ddb.createTable(createTableReq, function (err, data) {
-    if (err) {
-      debug('error creating table', err);
-      return deferred.reject(err);
-    }
-    debug('table created', data);
-    deferred.resolve(data);
-  });
-  return deferred.promise.nodeify(next);
+    debug('ddb.createTable request:', createTableReq);
 
+
+    ddb.createTable(createTableReq, function (err, data) {
+      if (err) {
+        debug('error creating table', err);
+        return reject(err);
+      }
+      debug('table created', data);
+      resolve(data);
+    });
+  })
 };
 
 Table.prototype.delete = function (next) {
   let deleteTableReq = {
     TableName: this.name
   };
 
-  debug('ddb.deleteTable request:', deleteTableReq);
-
-  let ddb = this.base.ddb();
+  if(next){
+    this.delete()
+      .then(result => {next(undefined,result)}, err => {next(err)})
+    return
+  }
 
-  const deferred = Q.defer();
+  return new Promise((resolve, reject) => {
+    debug('ddb.deleteTable request:', deleteTableReq);
 
-  ddb.deleteTable(deleteTableReq, function (err, data) {
-    if (err) {
-      debug('error deleting table', err);
-      return deferred.reject(err);
-    }
-    debug('deleted table', data);
-    deferred.resolve(data);
-  });
+    let ddb = this.base.ddb();
 
-  return deferred.promise.nodeify(next);
 
+    ddb.deleteTable(deleteTableReq, function (err, data) {
+      if (err) {
+        debug('error deleting table', err);
+        return reject(err);
+      }
+      debug('deleted table', data);
+      resolve(data);
+    });
+  })
 };
 
 Table.prototype.update = function (next) {
   // let ddb = this.base.ddb();
   // ddb.updateTable();
-  const deferred = Q.defer();
-  deferred.reject(new Error('TODO'));
```

Comment:
I wonder who wrote this code, and what the purpose of this is. Before this gets merged in, I should probably figure out how to look at the blame, to figure out when this was written to track it down and figure out how to implement it.

If you are able to do that quickly that'd be awesome. But I think it needs to be done before this gets merged in, or else I think it'd be a lot harder afterwards.